### PR TITLE
Fixed local HTTPS/CAC Authentication to work with the Terraform AWS Deployment Changes

### LINKED ISSUE #114: Configure automated Local Build to Work with AWS auto-deployment adjustments
**Is your feature request related to a problem? Please describe.**
As a developer, I would like to be able to run a local HTTPS build using the exact container configurations used for AWS auto-deployment.

**Describe the solution you'd like**
A bash script which can modify DNS records in the devs hosts file, generate self signed keys, and push them into their .env on a docker-compose command.

**Describe alternatives you've considered**
Directions could be written for the dev to do this manually.

**Additional context**
N/A.

### PR DESCRIPTION
**Pull request by dds-hansolo**
Thanks for submitting a pull request! Below are a few things you can do to help us more quickly review your changes.

## Checklist
I have…
- [X] summarized below my changes and noted which issues (if any) this pull request fixes or addresses.
- [X] thoroughly outlined below the steps necessary to test my changes.
- ~~[ ] attached screenshots illustrating relevant behavior before and after my changes.~~
- ~~[ ] review and update SSAG documentation if needed.~~

## Summary of Changes
This pull request…
- closes #114 
- enables developers to locally run a HTTPS/CAC enabled build/iterations, and streamlines the setup with a script.

## Testing
To verify the changes proposed in this pull request…
1. Pull down the branch
2. Follow the instructions to CONTRIBUTING.md
3. verify the website is accessible.
## Screenshots
~~_Attach relevant before and after screenshots here._~~

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -92,22 +92,15 @@ A docker-compose.yml file is included in the repository that quickly and easily
...
-  3. Generate self-signed public and private keys
-      - `openssl req -x509 -nodes -days 365 -newkey rsa:2048 -keyout ./nginx-selfsigned.key -out ./nginx-selfsigned.crt`
-      - `openssl dhparam -out ./dhparam.pem 1024`
-  4. Tell Chrome to allow self-signed certificates on localhost
-      - Navigate to `chrome://flags/#allow-insecure-localhost` and enable that setting
-  5. Execute docker-compose
+  1. After setting up the http environment, Navigate to solo/deploy/compose, make configureLocalEnv.sh executable, and run the script, set the Country name to US when prompted, skip the rest.
+      - `chmod +x configureLocalEnv.sh`
+      - `sudo sh configureLocalEnv.sh`
+      - `US` -> `ENTER` x6
```
Comment (jhackshaw-dds) [inline CONTRIBUTING.md]: Is the chmod +x necessary if the file is committed as executable? Is there a way to script the config prompts for cert generation (maybe like [this](https://unix.stackexchange.com/a/104305))?

Comment (dds-hansolo) [reply CONTRIBUTING.md]: These are great edits, see the recent commit.

File: deploy/compose/configure-local-environment.sh [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+# Remove generated keys
+rm ./nginx-selfsigned.key;
+rm ./nginx-selfsigned.crt;
+
+echo "local development environment setup has completed!\\n";
+echo "run the following command for HTTPS instantiation: docker-compose -f docker-compose.yml up --build";
+echo "run the following command for HTTP instantiation: docker-compose up --build\n";
+echo "solo https access point: https://stage.solo.localhost"
+echo "solo http access point: http://stage.solo.localhost:3000/"
```
Comment (jhackshaw-dds) [inline deploy/compose/configure-local-environment.sh:27]: This one could just be http://localhost:3000

File: deploy/compose/docker-compose.yml [hunk 3]
```diff
@@ -3,30 +3,30 @@ version: '3.4'
 services:
   db:
     image: postgres
-    env_file:
+    env_file:
```
Comment (jhackshaw-dds) [inline deploy/compose/docker-compose.yml:6]: Can you merge master back to get rid of the extra spaces here

File: CONTRIBUTING.md (hunk 1 above)
Comment (jhackshaw-dds) [reply CONTRIBUTING.md]: nicely done

File: deploy/compose/configure-local-environment.sh (hunk 2 above)
Comment (jhackshaw-dds) [reply deploy/compose/configure-local-environment.sh:27]: after talking, agree this is good

**Review by jhackshaw-dds — APPROVED** [review]

**Review by lit-dds — APPROVED** [review]
Changes reflect an easy way a developer can continue to locally test features.

**Pull request merged**